variables.tf

```diff
@@ -1,36 +1,67 @@
-########################################################################################################################
-# Input Variables
-########################################################################################################################
-
-#
-# Developer tips:
-#   - Below are some common module input variables
-#   - They should be updated for input variables applicable to the module being added
-#   - Use variable validation when possible
-#
-
-variable "name" {
+variable "region" {
   type        = string
-  description = "A descriptive name used to identify the resource instance."
+  description = "The IBM Cloud region where Cloud Monitoring instance will be created."
+  default     = "us-south"
+}
+
+variable "resource_group_id" {
+  type        = string
+  description = "The id of the IBM Cloud resource group where the Cloud Monitoring instance will be created."
+}
+
+variable "instance_name" {
+  type        = string
+  description = "The name of the IBM Cloud Monitoring instance to create. Defaults to 'cloud-monitoring-<region>'"
+  default     = null
 }
 
 variable "plan" {
   type        = string
-  description = "The name of the plan type supported by service."
-  default     = "standard"
+  description = "The IBM Cloud Monitoring plan to provision. Available: lite, graduated-tier"
+  default     = "lite"
+
   validation {
-    condition     = contains(["standard", "cos-one-rate-plan"], var.plan)
-    error_message = "The specified pricing plan is not available. The following plans are supported: 'standard', 'cos-one-rate-plan'"
+    condition     = can(regex("^lite$|^graduated-tier$", var.plan))
+    error_message = "The plan value must be one of the following: lite, graduated-tier."
   }
 }
 
-variable "resource_group_id" {
+variable "manager_key_name" {
   type        = string
-  description = "The ID of the resource group where you want to create the service."
+  description = "The name to give the IBM Cloud Monitoring manager key."
+  default     = "SysdigManagerKey"
+}
+
+variable "manager_key_tags" {
+  type        = list(string)
+  description = "Tags associated with the IBM Cloud Monitoring manager key."
+  default     = []
 }
 
 variable "resource_tags" {
   type        = list(string)
-  description = "List of resource tag to associate with the instance."
+  description = "Tags associated with the IBM Cloud Monitoring instance (Optional, array of strings)."
+  default     = []
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "Access Management Tags associated with the IBM Cloud Monitoring instance (Optional, array of strings)."
   default     = []
 }
+
+variable "enable_platform_metrics" {
+  type        = bool
+  description = "Receive platform metrics in the provisioned IBM Cloud Monitoring instance."
```

Comment:
Suggest to mention that only 1 instance in a given region can be enabled for platform metrics.